src/supernova/circuit.rs

```diff
@@ -41,15 +37,127 @@ use bellpepper_core::{
 use bellpepper::gadgets::Assignment;
 
 use abomonation_derive::Abomonation;
-use ff::Field;
+use ff::{Field, PrimeField};
 use itertools::Itertools as _;
 use serde::{Deserialize, Serialize};
+use std::marker::PhantomData;
 
 use crate::supernova::{
   num_ro_inputs,
   utils::{get_from_vec_alloc_relaxed_r1cs, get_selector_vec_from_index},
 };
 
+/// A helper trait for a step of the incremental computation for `SuperNova` (i.e., circuit for F) -- to be implemented by
+/// applications.
+pub trait StepCircuit<F: PrimeField>: Send + Sync + Clone {
```

Comment:
This stuff has been just removed without introducing changes, right?

Reply:
It has been moved from src/traits/circuit_supernova.rs